remote-index.md

```diff
@@ -0,0 +1,150 @@
+# Remote index
+
+## Motivation
+
+Clangd uses global idex for code completion, navigation and other useful
+features. Global index is typically generated automatically by background
+index at Clangd startup and kept fresh or manually by `clangd-indexer` (see
+[indexing](design/indexing.md) for more information). For large projects
+(e.g. LLVM and Chromium) index takes a long time to build (3-4 hours on
+powerful very machines for Chrome-sized project) and induces a large memory
+overhead (up to 6GB on RAM) to serve within clangd.
+
+We want to improve user experience to for developes of large projects. Hence,
+we introduce remote index: a feature that allows serving index on a separate
+machine and connecting to this index from your device. This means you don't
+have to build the index yourself anymore, clangd will use significantly less
+memory on the machine you are working on. You can work from your laptop or
+just a machine not powerful enough to process the whole codebase you work on
+and still get all clangd features that rely on indexing the whole project.
+
+Remote index feature is in beta now and you can expect minor bugs and
+changes. If you encounter any problems or have any questions, please reach
+out to us through Bug Tracker, Forum or Chat. Your feedback helps us polish
+the feature and make it more useful for developers using clangd.
+
+## Getting remote index support in clangd
```

Comment:
i think we should re-arrange the rest as:

- How it works
  - Building clangd-index-server and clangd with remote-index support
  - How to point client at the server

Afterwards, I believe "hosting the server" section should no longer be valid. It is too vague without details, and how it works section in theory should cover requirements for running a server from a high-level point of view.